### PR TITLE
Code quality (formating and flake8)

### PR DESCRIPTION
**Pull request by FFY00**
Okay, this has been a long time coming. The current code style is a mess, not even the indentation is consistent.

This PR introduces the following tools:
  - yapf (code formater)
  - isort (import sorter)
  - flake8 (various code quality checks)
  - pre-commit (misc checks, runs the tools listed above)

It also introduces a pre-commit config. pre-commit is used to run all the tests, it can be run standalone (`pre-commit run` or `pre-commit run --all-files`, I have this aliased to `pca` in my system) and it can also install a git hook to run everything when commiting. By default, it only runs the tools against the change files, this can be changed by passing the `--all-files` flag. To install the git hook, simply run `pre-commit install`. More info can be found here: https://pre-commit.com.

I also added a github action workflow to run all the checks.

I opted for yapf instead of black as I think black is very strict and has made some questionable choices when it comes to the code style they enforce.

@pfps what do you think?

### CODE REVIEW

**Comment by FFY00** [comment]
@pfps if you prefer black, we can use it instead.

**Comment by pfps** [comment]
@FFY00 I'm not a fan of black over yapf so yapf is fine by me.

We do want to be able to get `git blame` to overlook this commit (or commits).  There is a way to do this, but I forget where I saw how to do it.

**Comment by pfps** [comment]
I found something on ignoring a commit in git blame https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame
I tried it out and it works fine.  To make it work install a .git-blame-ignore-revs file with the 40-char commit ID and 
```
git config blame.ignoreRevsFile .git-blame-ignore-revs
```
I guess this has to be done after merging this PR to get the final IDs.

**Comment by pfps** [comment]
I had to
```
pip install --user pre-commit
```
to get this all to work. I wonder where this should be mentioned.

**Comment by FFY00** [comment]
I just pushed a `.git-blame-ignore-revs`.

```
git blame --ignore-revs-file .git-blame-ignore-revs
```

**Comment by FFY00** [comment]
> I had to
> 
> ```
> pip install --user pre-commit
> ```
> 
> to get this all to work. I wonder where this should be mentioned.

I don't think we need to specify how to install pre-commit. You install it the same way you install any other python package.

**Comment by pfps** [comment]
OK, I didn't realize that it was a python package.

**Comment by FFY00** [comment]
@pfps so is everything okay with you? Should we merge?

**Comment by pfps** [comment]
I downloaded and tried to run and encountered a problem:
```
idefix Solaar> bin/solaar
solaar: error: Traceback (most recent call last):
  File "/home/local/SoftwareDownloads/Solaar/lib/solaar/gtk.py", line 121, in main
    import solaar.ui as ui
  File "/home/local/SoftwareDownloads/Solaar/lib/solaar/ui/__init__.py", line 29, in <module>
    from . import notify, tray, window
  File "/home/local/SoftwareDownloads/Solaar/lib/solaar/ui/tray.py", line 35, in <module>
    from .window import popup as _window_popup
  File "/home/local/SoftwareDownloads/Solaar/lib/solaar/ui/window.py", line 34, in <module>
    from solaar.ui import ui_async as _ui_async
ImportError: cannot import name 'ui_async' from partially initialized module 'solaar.ui' (most likely due to a circular import) (/home/local/SoftwareDownloads/Solaar/lib/solaar/ui/__init__.py)
```
I'm not sure what is causing this.

**Comment by pfps** [comment]
Otherwise good.  I like the code style.

**Comment by pfps** [comment]
@FFY00 What's the status of the import error in this PR?

**Comment by FFY00** [comment]
I have not looked into it yet. I remember splitting up some imports, maybe that cause this.

File: lib/solaar/ui/__init__.py [hunk 1]
```diff
@@ -88,8 +92,6 @@ def ui_async(function, *args, **kwargs):
 #
 #
 
-from . import notify, tray, window
-
```
Comment (pfps) [inline lib/solaar/ui/__init__.py:87]: window uses ui_async so this import can't be moved up in the file

**Comment by pfps** [comment]
@FFY00 I think I found the problem - it's in lib/solaar/ui/__init__.py

File: lib/solaar/ui/__init__.py (hunk 1 above)
Comment (pfps) [reply lib/solaar/ui/__init__.py:87]: @FFY00 To keep the style cops happy, the import needs to be hidden in an `if True:`
I tried to figure out how to refactor, but it was becoming quite a mess and this is much easier for now.

**Comment by pfps** [comment]
@FFY00 Getting this cleaned up and merged soon would be very useful.

**Comment by FFY00** [comment]
Sorry for the delay! I have been busy.

File: lib/solaar/ui/__init__.py (hunk 1 above)
Comment (FFY00) [reply lib/solaar/ui/__init__.py:87]: I added `# isort:skip` to make isort skip the line and `# noqa: E402` to make flake8 ignore `E402` (`module level import not at top of file`).

**Comment by FFY00** [comment]
Can you do a quick sweep through the code to make sure we didn't pull up other import statements like this. It's a runtime error so we only run into it when we actually run the code.

**Comment by pfps** [comment]
I checked all the major pieces, both ui and cli, and everything appears OK, with this change.

I did find a bug in the config command, though.

**Comment by pfps** [comment]
@FFY00 I checked all the major pieces, both ui and cli, and everything appears OK, with this change.

I did find a bug in a descriptor, though - the M510 doesn't have smooth scroll.  I'll fix as soon as this PR is fixed and merged.

**Comment by FFY00** [comment]
Ack, so should we merge it?

**Comment by pfps** [comment]
@FFY00 Yes, I say go ahead.

File: lib/solaar/ui/__init__.py (hunk 1 above)
Comment (pfps) [reply lib/solaar/ui/__init__.py:87]: OK

**Comment by pfps** [comment]
@FFY00 It looks as if the commit ids change when the commit is merged so the ids in .git-blame-ignore-revs need to be changed.

**Comment by FFY00** [comment]
Oh, I thought we were in sync with master.

**Pull request merged**